shared/msg_conversions/tools/landmark_msg_cnv.cc

```diff
@@ -1,24 +1,59 @@
-// Copyright 2015 Intelligent Robotics Group, NASA ARC
-
-#include <ff_msgs/VisualLandmarks.h>
+/* Copyright (c) 2017, United States Government, as represented by the
+ * Administrator of the National Aeronautics and Space Administration.
+ *
+ * All rights reserved.
+ *
+ * The Astrobee platform is licensed under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with the
+ * License. You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
 
 #include <ff_common/init.h>
 #include <ff_common/utils.h>
+#include <ff_common/ros.h>
+
 #include <gflags/gflags.h>
-#include <ros/ros.h>
+
+#ifdef ROS1
+#include <ff_msgs/VisualLandmarks.h>
 #include <sensor_msgs/PointCloud2.h>
 
+#else
+#include <ff_msgs/msg/visual_landmarks.hpp>
+namespace ff_msgs {
+typedef msg::VisualLandmarks VisualLandmarks;
+typedef msg::VisualLandmarks::SharedPtr VisualLandmarksPtr;
+}  // namespace ff_msgs
+#include <sensor_msgs/msg/point_cloud2.hpp>
+namespace sensor_msgs {
+typedef msg::PointCloud2 PointCloud2;
+}
+namespace std_msgs {
+typedef msg::Header Header;
+}
+
+#endif
+
 DEFINE_string(input_topic, "/loc/ml/features",
               "The input features topic to convert.");
 DEFINE_string(output_topic, "/loc/rviz",
               "The output topic suitable for rviz.");
 
-ros::Publisher landmarks_pub;
+Publisher<sensor_msgs::PointCloud2> landmarks_pub = NULL;
+// auto landmarks_pub;
 
-void landmarks_callback(ff_msgs::VisualLandmarksPtr const & l) {
+void landmarks_callback(ff_msgs::VisualLandmarksPtr const l) {
   sensor_msgs::PointCloud2 out;
   out.header = std_msgs::Header();
-  out.header.stamp = ros::Time::now();
+  out.header.stamp = ROS_TIME_NOW();
```

Comment:
This is another case using a macro that looks different from the idiomatic ROS2 API call, which I guess would be `node->getClock()->now()` (meaning you need to have the node pointer handy).

If the definition of `ROS_TIME_NOW()` is similar to the example in `ros_compat.h`, it's mildly concerning. Their `ROS_CREATE_NODE()` macro defines its own private global clock object to be used by future `ROS_TIME_NOW()` calls. This defeats the ROS2 concept, as I understand it, that each node has its own clock. I'm not clear how important that compartmentalization is, but if ROS2 supports running multiple nodes in the same process, it might be important to use the clock from the right node (e.g., some nodes might have their clocks initialized before others).

I think the namespaced inline function approach can't provide a definition that looks exactly like the ROS2 API call since the API involves a member function, but if we relax the requirement to match the syntax exactly, at least it could provide an inline function `ff_ros::now(node)` implemented as:
- ROS1: `ros::Time::now()` (ignoring the node argument)
- ROS2: `node->getClock()->now()`

Or we could decide to just stick with a macro for now, at the risk of having unexpected hassles later.

But if we are using macros for ROS abstraction, they should have a standard prefix such as `FF_ROS`...